### PR TITLE
Bugfix status cache

### PR DESCRIPTION
**Pull request by nessiLahav**
### What does this PR do?
Fix cache bug in authn-jwt status flow

### What ticket does this PR close?
Resolves #[relevant GitHub issues, eg 76]

#### Change log
- [ ] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [ ] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [ ] This PR does not require updating any documentation

#### API Changes
- [ ] The [OpenAPI spec](https://github.com/cyberark/conjur-openapi-spec) has been updated to meet new API changes (or an issue has been opened), or
- [ ] The changes in this PR do not affect the Conjur API

### CODE REVIEW

File: app/domain/authentication/authn_jwt/validate_status.rb [hunk 1]
```diff
@@ -3,6 +3,16 @@ module AuthnJwt
 
     ValidateStatus = CommandClass.new(
       dependencies: {
+        fetch_signing_key: ::Util::ConcurrencyLimitedCache.new(
```
Comment (sashaCher) [inline app/domain/authentication/authn_jwt/validate_status.rb:6]: I think that it worth to align the parameter name in different classes https://github.com/cyberark/conjur/blob/c4eedc7b95ee1fb5bd5a9581300614f13bbe880c/app/domain/authentication/authn_jwt/validate_and_decode/validate_and_decode_token.rb#L10
CC: @tzheleznyak

File: app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb [hunk 2]
```diff
@@ -105,6 +105,10 @@ def create_signing_key_interface
           @create_signing_key_interface ||= Authentication::AuthnJwt::SigningKey::CreateSigningKeyFactory.new
         end
 
+        def fetch_jwt_claims_to_validate
+          @fetch_jwt_claims_to_validate ||= ::Authentication::AuthnJwt::ValidateAndDecode::FetchJwtClaimsToValidate.new
+        end
+
```
Comment (sashaCher) [inline app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb:111]: How is it related?

File: app/domain/authentication/authn_jwt/validate_status.rb [hunk 3]
```diff
@@ -134,25 +144,15 @@ def webservice
...
+          cache_key: signing_key_interface.signing_key_uri,
+          signing_key_interface: signing_key_interface
         )
+        @logger.debug(LogMessages::Authentication::AuthnJwt::ValidatedSigningKeyConfiguration.new)
       end
-
-      def fetch_signing_key_interface
-        @fetch_signing_key_interface ||= @create_signing_key_interface.call(
+      
+      def signing_key_interface
```
Comment (sashaCher) [inline app/domain/authentication/authn_jwt/validate_status.rb:154]: Is it aligned in behaviour with the next remark from the main PR
https://github.com/cyberark/conjur/blob/c4eedc7b95ee1fb5bd5a9581300614f13bbe880c/app/domain/authentication/authn_jwt/validate_and_decode/validate_and_decode_token.rb#L46

File: app/domain/authentication/authn_jwt/validate_status.rb [hunk 4]
```diff
@@ -134,25 +144,15 @@ def webservice
...
-          ::Util::RateLimitedCache.new(
-            fetch_signing_key_interface,
-            refreshes_per_interval: ::Authentication::AuthnJwt::CACHE_REFRESHES_PER_INTERVAL,
-            rate_limit_interval: ::Authentication::AuthnJwt::CACHE_RATE_LIMIT_INTERVAL,
-            logger: @logger
-          ),
-          max_concurrent_requests: ::Authentication::AuthnJwt::CACHE_MAX_CONCURRENT_REQUESTS,
-          logger: @logger
+        @fetch_signing_key.call(
+          cache_key: signing_key_interface.signing_key_uri,
```
Comment (sashaCher) [inline app/domain/authentication/authn_jwt/validate_status.rb:148]: May be it's better to call `signing_key_interface` explicit once and then approach the member it creates

File: app/domain/authentication/authn_jwt/validate_status.rb (hunk 1 above)
Comment (tzheleznyak) [reply app/domain/authentication/authn_jwt/validate_status.rb:6]: @nessiLahav  rebase the bugfix branch to you branch

File: cucumber/authenticators_jwt/features/authn_status_jwt.feature [hunk 5]
```diff
@@ -5,7 +5,8 @@ Feature: JWT Authenticator - Status Check
 
   @sanity
   Scenario: ONYX-9122: A valid JWT status request, 200 OK
-    Given I load a policy:
+    Given I initialize JWKS endpoint with file "myJWKs.json"
```
Comment (tzheleznyak) [inline cucumber/authenticators_jwt/features/authn_status_jwt.feature:8]: Use remote endpoint like the other tests

Comment (nessiLahav) [reply cucumber/authenticators_jwt/features/authn_status_jwt.feature:8]: Why?

File: app/domain/authentication/authn_jwt/validate_status.rb (hunk 3 above)
Comment (nessiLahav) [reply app/domain/authentication/authn_jwt/validate_status.rb:154]: @sashaCher Class is created per request so the members under the class like `@signing_key_interface` are not static.
What is static is the dependencies of the command class which there the cache is defined.

@tzheleznyak Do you agree ?

Comment (tzheleznyak) [reply app/domain/authentication/authn_jwt/validate_status.rb:154]: lets talk about it

**Review by sashaCher — APPROVED** [review]
Let it be

**Pull request merged**